fakeredis/commands_mixins/json_mixin.py

```diff
@@ -0,0 +1,605 @@
+
+        For more information see `JSON.DEL
+        <https://redis.io/commands/json.del>`_.
+        """
+        cached_value = json.loads(key.value or b"null")
+
+        if cached_value is None:
+            return 0
+
+        path = format_jsonpath(path or b"$")
+
+        if path != "$":
+            raise NotImplementedError("Path-based key-value deletion not yet supported!")
+
+        key.value = None
+        key.writeback()
+
+        return 1
+
+    # `forget` is an alias for `delete`
+    json_forget = json_del
+
+    @command(
+        name="JSON.GET",
+        fixed=(Key(),),
+        repeat=(bytes,),
+    )
+    def json_get(
+        self: FakeSocket,
+        name: CommandItem,
+        *args: bytes,
+    ) -> bytes:
+        """Get the object stored as a JSON value at key `name`.
+
+        `args` is zero or more paths, and defaults to root path
+        ``no_escape` is a boolean flag to add no_escape option to get
+        non-ascii characters
+
+        For more information see `JSON.GET <https://redis.io/commands/json.get>`_.
+        """
+        no_escape, args = (
+            bool(b"noescape" in args),
+            tuple(
+                map(
+                    parse_jsonpath,
+                    (format_jsonpath(arg) for arg in args if arg != b"noescape"),
+                )
+            ),
+        )
+
+        if no_escape:
+            # Silently ignore this. It was apparently originally added
+            # to RedisJSON to maintain compatibility with ReJSON v1.0
+            # which has this option. Per the notes associated with
+            # Issue #145 of RedisJSON's public git repository, `no_escape`
+            # is the implicit behavior always in current and future
+            # versions of the module. For details, see:
+            # https://github.com/RedisJSON/RedisJSON/issues/145
+            pass
+
+        path_count = len(args)
+        cached_value = json.loads(name.value)
+
+        callers = starmap(
+            methodcaller,
+            zip(
+                repeat("find", path_count),
+                repeat(cached_value, path_count),
+            ),
+        )
+        path_values = tuple(
+            map(
+                attrgetter("value"),
+                chain.from_iterable(
+                    (
+                        caller(path)
+                        for caller, path in zip(
+                            callers,
+                            args,
+                        )
+                    )
+                ),
+            )
+        )
+
+        if len(args) == 1:
+            path_values = path_values[0]
+
+        return json.dumps(path_values).encode()
+
+    @command(
+        name="JSON.SET",
+        fixed=(Key(), bytes, JSONObject),
+        repeat=(bytes,),
+    )
+    def json_set(
+        self: FakeSocket,
+        name: CommandItem,
+        path: bytes,
+        obj: JsonType,
+        flag: Optional[Literal[b"NX", b"XX"]] = None,
```

Comment:
See how NX/XX parameters are handled in various commands, example (generic_mixin.py -> _expireat)

Reply:
Resolved via TDD against the upstream `redis-py` test suite.